internal/internal.go

```diff
@@ -0,0 +1,33 @@
+package internal
+
+import (
+	cid "github.com/ipfs/go-cid"
+	cbg "github.com/whyrusleeping/cbor-gen"
+)
+
+const (
+	// Width must be a power of 2. We set this to 8.
+	MaxIndexBits = 63
+	WidthBits    = 3
+	Width        = 1 << WidthBits             // 8
+	BitfieldSize = 1                          // ((width - 1) >> 3) + 1
```

Comment:
Also if there is someway to document that the magic number 3 here is log of number of bits in a byte and unrelated to WidthBits it may help preempt some confusion.

Reply:
When this PR is merged I'll rebase and clean up my big docs PR #23, make it lighter-touch and easier to merge as _just_ docs (currently has a couple of minor code changes to help with making it reader-friendly). That should help with making a lot of these magic values clearer.

Reply:
IIRC, we thought the math was too confusing and opaque so we just asserted it. I'm going to clean this all up in my PR to make the width configurable.